concepts/functions/about.md

```diff
@@ -6,60 +6,82 @@ Python, like other programming languages, has [_built-in functions_][build-in fu
-Data. known as `parameters`, can be passed to the function by placing them inside the parenthesis. A broader term for parameters is `arguments`. Functions can perform different tasks depending on the value of the parameters.
+To execute the code inside a function, you need to call the function, which is done by using the function name followed by parenthesese [`()`].
+Data. known as [`arguments`][arguments], can be passed to the function by placing them inside the parenthesese.
+In the function definition any arguments are defined by [`parameters`][parameters].
+Functions can perform different tasks depending on the arguments passed to the parameters.
 
-A function can also return a value using the `return` keyword. The value returned by the function is known as the `return value`. The return value is returned to the caller of the function.
+A function can also return a value using the [`return`][return] keyword.
+The value returned by the function is known as the `return value`.
+The return value is returned to the caller of the function.
 
 ## Creation
 
-In python, functions are created using the `def` keyword. The function definition is followed by the function name and parenthesis [`()`]. Inside the parenthesis, the parameters are specified, separated by commas. After the close parenthesis, the colon (`:`) is used to separate the function signature from the function body.
+In python, functions are created using the [`def`][def] keyword.
+The function definition is followed by the function name and parentheses [`()`].
+Inside the parentheses, the parameters are specified, separated by commas.
+After the close parenthesis, the colon (`:`) is used to separate the function signature from the function body.
 
-The function body is a block of code that is executed when the function is called. The body of the function is indented. The indentation is important because Python relies on it to know where that block of code ends. A value can be returned from the function by using the `return` keyword, which can then be used by the caller of the function.
+The function body is a block of code that is executed when the function is called.
+The body of the function is indented.
+The indentation is important because Python relies on it to know where that block of code ends.
+A value can be returned from the function by using the `return` keyword, which can then be used by the caller of the function.
 
 ```python
 def function_name(parameter1, parameter2, ...):
     # function body
-    return value
+    return parameter1 + parameter2
+
 ```
 
 We can also define a function without any parameters or return value.
 
 ```python
 def function_name():
     # function body
+    pass
+
 ```
 
+Note that the function does need a body, even if the body does nothing, or trying to run the program will generate an indentation error.
```

Comment:
Maybe add an example of the error here?  You can't make it happen in the REPL, but when in happens in code, it's fairly inscrutable because all you get is a line number and a random-looking carrot with the error message.  It doesn't even manage to print the function definition.  😱 :

```python
File ~/temp.py:66
    
    ^
IndentationError: expected an indented block
```